2019-05-31 18:26:10 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "CREATE TABLE HUONE(NUMERO INTEGER AS PRIMARY[*] KEY, TYYPPI VARCHAR(25), PAIVAHINTA INTEGER, VARATTU BOOLEAN); "; expected "NOT, EXISTS, INTERSECTS, SELECT, FROM, WITH"; SQL statement:
CREATE TABLE Huone(numero Integer as primary key, tyyppi varchar(25), paivahinta Integer, varattu Boolean); [42001-197]
2019-06-02 14:13:05 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT  FROM HUONE [*]"; expected "., (, USE, AS, RIGHT, LEFT, FULL, INNER, JOIN, CROSS, NATURAL, ,, SELECT"; SQL statement:
SELECT  FROM Huone [42001-197]
2019-06-02 15:30:04 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "ASIAKAS" not found; SQL statement:
CREATE TABLE Varaus(id Integer Primary key, asiakas_id Integer, alkupvm Date,loppupvm Date, onkoLisavarusteita Boolean, kestoPaivina Integer, huoneidenLKM Integer, yhteishinta Integer, Foreign key(asiakas_id) References Asiakas(id)) [42102-197]
2019-06-02 15:31:23 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "CREATE TABLE ASIAKAS(ID INTEGER PRIMARY KEY, NIMI VARCHAR(30), PUHELINNRO VARCHAR(15), EMAIL VARCHAR(35),  [*]"; expected "identifier"; SQL statement:
CREATE TABLE Asiakas(id Integer Primary key, nimi VarChar(30), puhelinnro VarChar(15), email VarChar(35),  [42001-197]
2019-06-02 15:32:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "CREATE TABLE ASIAKAS(ID INTEGER PRIMARY KEY, NIMI VARCHAR(30), PUHELINNRO VARCHAR(15), EMAIL VARCHAR(35);[*] "; expected "FOR, UNSIGNED, INVISIBLE, VISIBLE, NOT, NULL, AS, DEFAULT, GENERATED, ON, NOT, NULL, AUTO_INCREMENT, BIGSERIAL, SERIAL, IDENTITY, NULL_TO_DEFAULT, SEQUENCE, SELECTIVITY, COMMENT, CONSTRAINT, PRIMARY, UNIQUE, NOT, NULL, CHECK, REFERENCES, ,, )"; SQL statement:
CREATE TABLE Asiakas(id Integer Primary key, nimi VarChar(30), puhelinnro VarChar(15), email VarChar(35); [42001-197]
2019-06-02 15:34:36 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "CREATE TABLE ASIAKAS(ID INTEGER PRIMARY KEY, NIMI VARCHAR(30), PUHELINNRO VARCHAR(15), EMAIL VARCHAR(35);[*] "; expected "FOR, UNSIGNED, INVISIBLE, VISIBLE, NOT, NULL, AS, DEFAULT, GENERATED, ON, NOT, NULL, AUTO_INCREMENT, BIGSERIAL, SERIAL, IDENTITY, NULL_TO_DEFAULT, SEQUENCE, SELECTIVITY, COMMENT, CONSTRAINT, PRIMARY, UNIQUE, NOT, NULL, CHECK, REFERENCES, ,, )"; SQL statement:
CREATE TABLE Asiakas(id Integer Primary key, nimi VarChar(30), puhelinnro VarChar(15), email VarChar(35); [42001-197]
2019-06-02 16:15:23 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Unknown data type: "BOOELAN"; SQL statement:
CREATE TABLE Lisavarusteet(id Integer, varaus_id Integer, payTV Boolean, minibaari Boolean, sukset Booelan, Foreign key(varaus_id) References Varaus(id)) [50004-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.command.Parser.parseColumnWithType(Parser.java:4400)
	at org.h2.command.Parser.parseColumnForTable(Parser.java:4215)
	at org.h2.command.Parser.parseTableColumnDefinition(Parser.java:6662)
	at org.h2.command.Parser.parseCreateTable(Parser.java:6570)
	at org.h2.command.Parser.parseCreate(Parser.java:4615)
	at org.h2.command.Parser.parsePrepared(Parser.java:380)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:304)
	at varausjarjestelma.VarausjarjestelmaSovellus.alustaTietokanta(VarausjarjestelmaSovellus.java:38)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:19)
2019-06-03 16:28:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'q' (VARAUS: ASIAKAS_ID INTEGER)"; SQL statement:
INSERT INTO "PUBLIC"."VARAUS"("ID","ASIAKAS_ID","ALKUPVM","LOPPUPVM","ONKOLISAVARUSTEITA","KESTOPAIVINA","HUONEIDENLKM","YHTEISHINTA")VALUES(?,?,?,?,?,?,?,?) -- (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8) [22018-197]
2019-06-03 16:28:58 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'q' (VARAUS: ASIAKAS_ID INTEGER)"; SQL statement:
INSERT INTO "PUBLIC"."VARAUS"("ID","ASIAKAS_ID","ALKUPVM","LOPPUPVM","ONKOLISAVARUSTEITA","KESTOPAIVINA","HUONEIDENLKM","YHTEISHINTA")VALUES(?,?,?,?,?,?,?,?) -- (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8) [22018-197]
2019-06-03 16:32:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'' (VARAUS: ASIAKAS_ID INTEGER)"; SQL statement:
INSERT INTO "PUBLIC"."VARAUS"("ID","ASIAKAS_ID","ALKUPVM","LOPPUPVM","ONKOLISAVARUSTEITA","KESTOPAIVINA","HUONEIDENLKM","YHTEISHINTA")VALUES(?,?,?,?,?,?,?,?) -- (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8) [22018-197]
2019-06-03 16:32:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'' (VARAUS: ASIAKAS_ID INTEGER)"; SQL statement:
INSERT INTO "PUBLIC"."VARAUS"("ID","ASIAKAS_ID","ALKUPVM","LOPPUPVM","ONKOLISAVARUSTEITA","KESTOPAIVINA","HUONEIDENLKM","YHTEISHINTA")VALUES(?,?,?,?,?,?,?,?) -- (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8) [22018-197]
2019-06-27 09:23:16 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at varausjarjestelma.VarausjarjestelmaSovellus.alustaTietokanta(VarausjarjestelmaSovellus.java:32)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:19)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 26 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 20 more
2019-06-27 09:23:42 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:247)
	at varausjarjestelma.VarausjarjestelmaSovellus.alustaTietokanta(VarausjarjestelmaSovellus.java:32)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:19)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 26 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 20 more
2019-06-27 14:36:46 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "VARATTU" not found; SQL statement:
INSERT INTO Huone(numero, tyyppi, paivahinta, varattu)VALUES(?, ?, ?, ?) [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.table.Table.getColumn(Table.java:682)
	at org.h2.command.Parser.parseColumn(Parser.java:936)
	at org.h2.command.Parser.parseColumnList(Parser.java:920)
	at org.h2.command.Parser.parseInsertGivenTable(Parser.java:1258)
	at org.h2.command.Parser.parseInsert(Parser.java:1222)
	at org.h2.command.Parser.parsePrepared(Parser.java:431)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:304)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:311)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at org.springframework.jdbc.core.JdbcTemplate$SimplePreparedStatementCreator.createPreparedStatement(JdbcTemplate.java:1557)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:615)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:862)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:917)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:927)
	at varausjarjestelma.HuoneDao.create(HuoneDao.java:21)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaHuone(Tekstikayttoliittyma.java:71)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:44)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-06-27 14:38:57 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column count does not match; SQL statement:
INSERT INTO Huone(numero, tyyppi, paivahinta)VALUES(?, ?, ?, ?) [21002-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.message.DbException.get(DbException.java:144)
	at org.h2.command.dml.Insert.prepare(Insert.java:306)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:304)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:311)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at org.springframework.jdbc.core.JdbcTemplate$SimplePreparedStatementCreator.createPreparedStatement(JdbcTemplate.java:1557)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:615)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:862)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:917)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:927)
	at varausjarjestelma.HuoneDao.create(HuoneDao.java:21)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaHuone(Tekstikayttoliittyma.java:71)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:44)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-06-29 14:18:22 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "ASIASKAS" not found; SQL statement:
INSERT INTO Asiaskas(nimi, puhelinnro, email) VALUES(?,?,?) [42102-197]
2019-06-29 17:58:44 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO HUONE(NUMERO, TYYPPI, PAIVAHINTA) VALUES(101, 'Standard', 100 [*]"; expected "[, ::, *, /, %, +, -, ||, ~, !~, NOT, LIKE, ILIKE, REGEXP, IS, IN, BETWEEN, AND, OR, ,, )"; SQL statement:
INSERT INTO Huone(numero, tyyppi, paivahinta) VALUES(101, 'Standard', 100 [42001-197]
2019-06-30 11:10:30 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Cannot parse "DATE" constant "2019-09-09T16:00"; SQL statement:
INSERT INTO Varaus(alkupvm, loppupvm, asiakas_id,huone_numero, kestoPaivina)VALUES(?,?,?,?,?) -- (?1, ?2, ?3, ?4, ?5) [22007-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.value.ValueDate.parse(ValueDate.java:74)
	at org.h2.value.Value.convertTo(Value.java:1069)
	at org.h2.table.Column.convert(Column.java:177)
	at org.h2.command.dml.Insert.insertRows(Insert.java:166)
	at org.h2.command.dml.Insert.update(Insert.java:134)
	at org.h2.command.CommandContainer.update(CommandContainer.java:102)
	at org.h2.command.Command.executeUpdate(Command.java:261)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:199)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:153)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at org.springframework.jdbc.core.JdbcTemplate.lambda$update$1(JdbcTemplate.java:894)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:617)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:893)
	at varausjarjestelma.VarausDao.createAndReturnKey(VarausDao.java:26)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaVaraus(Tekstikayttoliittyma.java:201)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:56)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: java.lang.NumberFormatException: For input string: "09T16:00"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at org.h2.util.DateTimeUtils.parseDateValue(DateTimeUtils.java:349)
	at org.h2.value.ValueDate.parse(ValueDate.java:72)
	... 23 more
2019-06-30 22:13:04 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "NAME" not found; SQL statement:
SELECT * FROM VARAUS where name = 'Teppo Ruohonen' [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)
	at org.h2.expression.Comparison.optimize(Comparison.java:177)
	at org.h2.command.dml.Select.prepare(Select.java:861)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-06-30 22:13:17 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "NAME" not found; SQL statement:
SELECT * FROM asiakas where name = 'Teppo Ruohonen' [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)
	at org.h2.expression.Comparison.optimize(Comparison.java:177)
	at org.h2.command.dml.Select.prepare(Select.java:861)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-06-30 22:14:13 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT * FROM ASIAKAS WHERE EMAIL = ?'luukku.com'[*] "; SQL statement:
SELECT * FROM asiakas where email = ?'luukku.com' [42000-197]
2019-06-30 22:45:07 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Ambiguous column name "HUONE_NUMERO"; SQL statement:
select huone_numero from huone 
join huone_varaus on huone_varaus.huone_numero = huone.numero
join varaus on varaus.id = huone_varaus.varaus_id
where loppupvm > '2019-07-10' [90059-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.mapColumn(ExpressionColumn.java:127)
	at org.h2.expression.ExpressionColumn.mapColumns(ExpressionColumn.java:97)
	at org.h2.command.dml.Select.mapColumns(Select.java:1237)
	at org.h2.command.dml.Select.init(Select.java:826)
	at org.h2.command.Parser.parseSelect(Parser.java:1925)
	at org.h2.command.Parser.parsePrepared(Parser.java:463)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-06-30 23:02:14 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Cannot parse "DATE" constant "2002"; SQL statement:
select numero, tyyppi, paivahinta from huone 
join huone_varaus on huone_varaus.huone_numero = huone.numero
join varaus on varaus.id = huone_varaus.varaus_id
where loppupvm < 2019-07-10 [22007-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.value.ValueDate.parse(ValueDate.java:74)
	at org.h2.value.Value.convertTo(Value.java:1069)
	at org.h2.expression.Comparison.optimize(Comparison.java:204)
	at org.h2.expression.ConditionAndOr.optimize(ConditionAndOr.java:130)
	at org.h2.command.dml.Select.prepare(Select.java:861)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeQuery(JdbcStatement.java:78)
	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:439)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.VarausDao.listaaVapaatHuoneet(VarausDao.java:72)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaVaraus(Tekstikayttoliittyma.java:147)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:56)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: java.lang.IllegalArgumentException: 2002
	at org.h2.util.DateTimeUtils.parseDateValue(DateTimeUtils.java:345)
	at org.h2.value.ValueDate.parse(ValueDate.java:72)
	... 25 more
2019-07-01 17:52:11 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Cannot parse "DATE" constant "2002"; SQL statement:
select numero, tyyppi, paivahinta from huone 
            left join huone_varaus on huone_varaus.huone_numero = huone.numero
            join varaus on varaus.id = huone_varaus.varaus_id
            where loppupvm < 2019-07-10 [22007-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.value.ValueDate.parse(ValueDate.java:74)
	at org.h2.value.Value.convertTo(Value.java:1069)
	at org.h2.expression.Comparison.optimize(Comparison.java:204)
	at org.h2.command.dml.Select.prepare(Select.java:861)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: 2002
	at org.h2.util.DateTimeUtils.parseDateValue(DateTimeUtils.java:345)
	at org.h2.value.ValueDate.parse(ValueDate.java:72)
	... 16 more
2019-07-01 17:52:22 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Cannot parse "DATE" constant "2002"; SQL statement:
select numero, tyyppi, paivahinta from huone 
            join huone_varaus on huone_varaus.huone_numero = huone.numero
            join varaus on varaus.id = huone_varaus.varaus_id
            where loppupvm < 2019-07-10 [22007-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.value.ValueDate.parse(ValueDate.java:74)
	at org.h2.value.Value.convertTo(Value.java:1069)
	at org.h2.expression.Comparison.optimize(Comparison.java:204)
	at org.h2.expression.ConditionAndOr.optimize(ConditionAndOr.java:130)
	at org.h2.command.dml.Select.prepare(Select.java:861)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: 2002
	at org.h2.util.DateTimeUtils.parseDateValue(DateTimeUtils.java:345)
	at org.h2.value.ValueDate.parse(ValueDate.java:72)
	... 17 more
2019-07-01 18:05:20 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT NUMERO, TYYPPI, PAIVAHINTA FROM HUONE 
           HUONE_VARAUS ON[*] HUONE_VARAUS.HUONE_NUMERO = HUONE.NUMERO
           VARAUS ON VARAUS.ID = HUONE_VARAUS.VARAUS_ID
           WHERE VARAUS.LOPPUPVM < '2019-07-10' "; SQL statement:
select numero, tyyppi, paivahinta from huone 
           huone_varaus on huone_varaus.huone_numero = huone.numero
           varaus on varaus.id = huone_varaus.varaus_id
           where varaus.loppupvm < '2019-07-10' [42000-197]
2019-07-01 18:05:27 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT NUMERO, TYYPPI, PAIVAHINTA FROM HUONE 
           HUONE_VARAUS ON[*] HUONE_VARAUS.HUONE_NUMERO = HUONE.NUMERO
           VARAUS ON VARAUS.ID = HUONE_VARAUS.VARAUS_ID
           WHERE LOPPUPVM < '2019-07-10' "; SQL statement:
select numero, tyyppi, paivahinta from huone 
           huone_varaus on huone_varaus.huone_numero = huone.numero
           varaus on varaus.id = huone_varaus.varaus_id
           where loppupvm < '2019-07-10' [42000-197]
2019-07-01 18:05:38 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT NUMERO, TYYPPI, PAIVAHINTA FROM HUONE 
           HUONE_VARAUS ON[*] HUONE_VARAUS.HUONE_NUMERO = HUONE.NUMERO
           VARAUS ON VARAUS.ID = HUONE_VARAUS.VARAUS_ID
           WHERE LOPPUPVM < '2019-07-10' "; SQL statement:
select numero, tyyppi, paivahinta from huone 
           huone_varaus on huone_varaus.huone_numero = huone.numero
           varaus on varaus.id = huone_varaus.varaus_id
           where loppupvm < '2019-07-10' [42000-197]
2019-07-01 18:06:03 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT NUMERO, TYYPPI, PAIVAHINTA FROM HUONE 
           HUONE_VARAUS ON[*] HUONE_VARAUS.HUONE_NUMERO = HUONE.NUMERO
           VARAUS ON VARAUS.ID = HUONE_VARAUS.VARAUS_ID
           WHERE LOPPUPVM < '2019-07-10' "; SQL statement:
select numero, tyyppi, paivahinta from huone 
           huone_varaus on huone_varaus.huone_numero = huone.numero
           varaus on varaus.id = huone_varaus.varaus_id
           where loppupvm < '2019-07-10' [42000-197]
2019-07-03 17:14:05 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "NUMERO" must be in the GROUP BY list; SQL statement:
SELECT numero, tyyppi, paivahinta FROM Huone 
JOIN Huone_varaus ON Huone_varaus.huone_numero = Huone.numero
JOIN Varaus ON Varaus.id = Huone_varaus.varaus_id
WHERE Varaus.loppupvm < ? OR Varaus.alkupvm > ?GROUP BY Huone.paivahinta [90016-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.updateAggregate(ExpressionColumn.java:172)
	at org.h2.command.dml.Select.queryGroup(Select.java:350)
	at org.h2.command.dml.Select.queryWithoutCache(Select.java:628)
	at org.h2.command.dml.Query.queryWithoutCacheLazyCheck(Query.java:114)
	at org.h2.command.dml.Query.query(Query.java:371)
	at org.h2.command.dml.Query.query(Query.java:333)
	at org.h2.command.CommandContainer.query(CommandContainer.java:114)
	at org.h2.command.Command.executeQuery(Command.java:202)
	at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:114)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
	at org.springframework.jdbc.core.JdbcTemplate$1.doInPreparedStatement(JdbcTemplate.java:678)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:617)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:669)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:700)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:712)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:768)
	at varausjarjestelma.VarausDao.listaaVapaatHuoneet(VarausDao.java:76)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaVaraus(Tekstikayttoliittyma.java:145)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:54)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-03 17:15:36 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "NUMERO" must be in the GROUP BY list; SQL statement:
SELECT numero, tyyppi, paivahinta FROM Huone 
JOIN Huone_varaus ON Huone_varaus.huone_numero = Huone.numero
JOIN Varaus ON Varaus.id = Huone_varaus.varaus_id
WHERE Varaus.loppupvm < ? OR Varaus.alkupvm > ?GROUP BY paivahinta [90016-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.updateAggregate(ExpressionColumn.java:172)
	at org.h2.command.dml.Select.queryGroup(Select.java:350)
	at org.h2.command.dml.Select.queryWithoutCache(Select.java:628)
	at org.h2.command.dml.Query.queryWithoutCacheLazyCheck(Query.java:114)
	at org.h2.command.dml.Query.query(Query.java:371)
	at org.h2.command.dml.Query.query(Query.java:333)
	at org.h2.command.CommandContainer.query(CommandContainer.java:114)
	at org.h2.command.Command.executeQuery(Command.java:202)
	at org.h2.jdbc.JdbcPreparedStatement.executeQuery(JdbcPreparedStatement.java:114)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
	at org.springframework.jdbc.core.JdbcTemplate$1.doInPreparedStatement(JdbcTemplate.java:678)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:617)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:669)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:700)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:712)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:768)
	at varausjarjestelma.VarausDao.listaaVapaatHuoneet(VarausDao.java:76)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaVaraus(Tekstikayttoliittyma.java:145)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:54)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-03 17:17:09 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at org.h2.server.web.WebServer.getConnection(WebServer.java:733)
	at org.h2.server.web.WebApp.login(WebApp.java:954)
	at org.h2.server.web.WebApp.process(WebApp.java:211)
	at org.h2.server.web.WebApp.processRequest(WebApp.java:170)
	at org.h2.server.web.WebThread.process(WebThread.java:134)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 29 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 23 more
2019-07-03 17:17:12 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at org.h2.server.web.WebServer.getConnection(WebServer.java:733)
	at org.h2.server.web.WebApp.login(WebApp.java:954)
	at org.h2.server.web.WebApp.process(WebApp.java:211)
	at org.h2.server.web.WebApp.processRequest(WebApp.java:170)
	at org.h2.server.web.WebThread.process(WebThread.java:134)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 29 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 23 more
2019-07-03 17:40:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Parameter "#1" is not set; SQL statement:
SELECT numero, tyyppi, paivahinta FROM Huone
            JOIN Huone_varaus ON Huone_varaus.huone_numero = Huone.numero
            JOIN Varaus ON Varaus.id = Huone_varaus.varaus_id
            WHERE Varaus.loppupvm < ? OR Varaus.alkupvm > ?
            GROUP BY numero [90012-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.Parameter.checkSet(Parameter.java:81)
	at org.h2.command.Prepared.checkParameters(Prepared.java:170)
	at org.h2.command.CommandContainer.query(CommandContainer.java:113)
	at org.h2.command.Command.executeQuery(Command.java:202)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:227)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-07-04 18:04:20 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column count does not match; SQL statement:
SELECT numero, tyyppi, paivahinta FROM Huone
            JOIN Huone_varaus ON Huone_varaus.huone_numero = Huone.numero
            JOIN Varaus ON Varaus.id = Huone_varaus.varaus_id
            WHERE Varaus.loppupvm < '2019-07-10'
UNION
SELECT * FROM Huone
GROUP BY numero [21002-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.message.DbException.get(DbException.java:144)
	at org.h2.command.dml.SelectUnion.init(SelectUnion.java:315)
	at org.h2.command.Parser.parseSelect(Parser.java:1925)
	at org.h2.command.Parser.parsePrepared(Parser.java:463)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-07-04 18:04:50 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column count does not match; SQL statement:
SELECT numero, tyyppi, paivahinta FROM Huone
            JOIN Huone_varaus ON Huone_varaus.huone_numero = Huone.numero
            JOIN Varaus ON Varaus.id = Huone_varaus.varaus_id
UNION
SELECT * FROM Huone
GROUP BY numero [21002-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.message.DbException.get(DbException.java:144)
	at org.h2.command.dml.SelectUnion.init(SelectUnion.java:315)
	at org.h2.command.Parser.parseSelect(Parser.java:1925)
	at org.h2.command.Parser.parsePrepared(Parser.java:463)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-07-04 21:10:00 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKAS_ID,HUONE_NUMERO, KESTOPAIVINA)VALUES(?,?,?,?,?)UNIONUPDATE[*] HUONE SET VARATTU = TRUEWHERE NUMERO = ? "; SQL statement:
INSERT INTO Varaus(alkupvm, loppupvm, asiakas_id,huone_numero, kestoPaivina)VALUES(?,?,?,?,?)UNIONUPDATE Huone SET varattu = trueWHERE numero = ? [42000-197]
2019-07-04 21:14:37 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKAS_ID,
                            HUONE_NUMERO, KESTOPAIVINA)
                            VALUES('2019-09-09', '2019-09-19', 5, 103, 10)
                            UNION[*]
                            UPDATE HUONE SET VARATTU = TRUE
                            WHERE NUMERO = 103 "; SQL statement:
INSERT INTO Varaus(alkupvm, loppupvm, asiakas_id,
                            huone_numero, kestoPaivina)
                            VALUES('2019-09-09', '2019-09-19', 5, 103, 10)
                            UNION
                            UPDATE Huone SET varattu = true
                            WHERE numero = 103 [42000-197]
2019-07-06 01:45:11 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "VARAUSWHERE" not found; SQL statement:
SELECT * FROM VarausWHERE id = ? [42102-197]
2019-07-06 14:11:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "HUONEWHERE" not found; SQL statement:
SELECT * FROM HuoneWHERE huone_Numero = ? [42102-197]
2019-07-06 14:13:06 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "HUONEWHERE" not found; SQL statement:
SELECT * FROM HuoneWHERE huone_Numero = ? [42102-197]
2019-07-06 14:14:18 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "HUONE_NUMERO" not found; SQL statement:
SELECT * FROM Huone WHERE huone_Numero = ? [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)
	at org.h2.expression.Comparison.optimize(Comparison.java:177)
	at org.h2.command.dml.Select.prepare(Select.java:861)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:304)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:311)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at org.springframework.jdbc.core.JdbcTemplate$SimplePreparedStatementCreator.createPreparedStatement(JdbcTemplate.java:1557)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:615)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:669)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:700)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:712)
	at org.springframework.jdbc.core.JdbcTemplate.queryForObject(JdbcTemplate.java:790)
	at varausjarjestelma.HuoneDao.lueHuoneNumerolla(HuoneDao.java:42)
	at varausjarjestelma.HuoneDao.haeHuoneita(HuoneDao.java:88)
	at varausjarjestelma.Tekstikayttoliittyma.haeHuoneita(Tekstikayttoliittyma.java:109)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:52)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-07 12:34:10 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at org.h2.server.web.WebServer.getConnection(WebServer.java:733)
	at org.h2.server.web.WebApp.login(WebApp.java:954)
	at org.h2.server.web.WebApp.process(WebApp.java:211)
	at org.h2.server.web.WebApp.processRequest(WebApp.java:170)
	at org.h2.server.web.WebThread.process(WebThread.java:134)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 29 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 23 more
2019-07-07 21:14:31 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "HUONEVARAUS" not found; SQL statement:
INSERT INTO HuoneVaraus(huone_numero, varaus_id)VALUES(?,?) [42102-197]
2019-07-10 09:57:41 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "NIMI" not found; SQL statement:
SELECT nimi, puhelinnro, emailFROM Asiakas [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)
	at org.h2.command.dml.Select.prepare(Select.java:858)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeQuery(JdbcStatement.java:78)
	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:439)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.AsiakasDao.list(AsiakasDao.java:52)
	at varausjarjestelma.VarausDao.listaaVaraukset(VarausDao.java:93)
	at varausjarjestelma.Tekstikayttoliittyma.listaaVaraukset(Tekstikayttoliittyma.java:175)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:55)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-10 09:59:24 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "HUONE_NUMERO" not found; SQL statement:
SELECT huone_numero, varaus_idFROM Huonevaraus [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)
	at org.h2.command.dml.Select.prepare(Select.java:858)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeQuery(JdbcStatement.java:78)
	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:439)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.HuoneVarausDao.list(HuoneVarausDao.java:47)
	at varausjarjestelma.VarausDao.listaaVaraukset(VarausDao.java:95)
	at varausjarjestelma.Tekstikayttoliittyma.listaaVaraukset(Tekstikayttoliittyma.java:175)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:55)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-10 10:13:17 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:541)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:112)
	at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:157)
	at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:115)
	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:78)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:371)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.HuoneDao.list(HuoneDao.java:62)
	at varausjarjestelma.Tekstikayttoliittyma.listaaHuoneet(Tekstikayttoliittyma.java:81)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:49)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 45 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 39 more
2019-07-10 10:13:42 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Table "ASIAKASWHERE" not found; SQL statement:
SELECT * FROM AsiakasWHERE id = ? [42102-197]
2019-07-10 13:47:08 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at org.h2.server.web.WebServer.getConnection(WebServer.java:733)
	at org.h2.server.web.WebApp.login(WebApp.java:954)
	at org.h2.server.web.WebApp.process(WebApp.java:211)
	at org.h2.server.web.WebApp.processRequest(WebApp.java:170)
	at org.h2.server.web.WebThread.process(WebThread.java:134)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 29 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 23 more
2019-07-10 15:13:30 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "puhnro" not found [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.jdbc.JdbcResultSet.getColumnIndex(JdbcResultSet.java:3148)
	at org.h2.jdbc.JdbcResultSet.get(JdbcResultSet.java:3247)
	at org.h2.jdbc.JdbcResultSet.getString(JdbcResultSet.java:310)
	at com.zaxxer.hikari.pool.HikariProxyResultSet.getString(HikariProxyResultSet.java)
	at varausjarjestelma.Asiakas.<init>(Asiakas.java:28)
	at varausjarjestelma.AsiakasDao.lambda$list$0(AsiakasDao.java:53)
	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:94)
	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:61)
	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:440)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.AsiakasDao.list(AsiakasDao.java:52)
	at varausjarjestelma.VarausDao.listaaVaraukset(VarausDao.java:93)
	at varausjarjestelma.Tekstikayttoliittyma.listaaVaraukset(Tekstikayttoliittyma.java:176)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:55)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-12 14:46:49 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:541)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:112)
	at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:157)
	at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:115)
	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:78)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:371)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.VarausDao.list(VarausDao.java:63)
	at varausjarjestelma.HuoneDao.haeHuoneita(HuoneDao.java:69)
	at varausjarjestelma.Tekstikayttoliittyma.haeHuoneita(Tekstikayttoliittyma.java:103)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:51)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 46 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 40 more
2019-07-12 14:47:54 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:541)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:112)
	at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:157)
	at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:115)
	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:78)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:371)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.VarausDao.list(VarausDao.java:63)
	at varausjarjestelma.HuoneDao.haeHuoneita(HuoneDao.java:69)
	at varausjarjestelma.Tekstikayttoliittyma.haeHuoneita(Tekstikayttoliittyma.java:103)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:51)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 46 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 40 more
2019-07-12 14:48:22 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:541)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:112)
	at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:157)
	at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:115)
	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:78)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:371)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.VarausDao.list(VarausDao.java:63)
	at varausjarjestelma.HuoneDao.haeHuoneita(HuoneDao.java:69)
	at varausjarjestelma.Tekstikayttoliittyma.haeHuoneita(Tekstikayttoliittyma.java:103)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:51)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 46 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 40 more
2019-07-13 22:25:02 database: flush
org.h2.message.DbException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.message.DbException.convert(DbException.java:307)
	at org.h2.mvstore.db.MVTableEngine$1.uncaughtException(MVTableEngine.java:95)
	at org.h2.mvstore.MVStore.handleException(MVStore.java:2506)
	at org.h2.mvstore.MVStore.panic(MVStore.java:383)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:364)
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:2934)
	at org.h2.mvstore.db.MVTableEngine$Store.open(MVTableEngine.java:155)
	at org.h2.mvstore.db.MVTableEngine.init(MVTableEngine.java:100)
	at org.h2.engine.Database.getPageStore(Database.java:2538)
	at org.h2.engine.Database.open(Database.java:709)
	at org.h2.engine.Database.openDatabase(Database.java:286)
	at org.h2.engine.Database.<init>(Database.java:280)
	at org.h2.engine.Engine.openSession(Engine.java:66)
	at org.h2.engine.Engine.openSession(Engine.java:179)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:157)
	at org.h2.engine.Engine.createSession(Engine.java:140)
	at org.h2.engine.Engine.createSession(Engine.java:28)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:351)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:124)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:103)
	at org.h2.Driver.connect(Driver.java:69)
	at com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:136)
	at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:369)
	at com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:198)
	at com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:467)
	at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:541)
	at com.zaxxer.hikari.pool.HikariPool.<init>(HikariPool.java:115)
	at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:112)
	at org.springframework.jdbc.datasource.DataSourceUtils.fetchConnection(DataSourceUtils.java:157)
	at org.springframework.jdbc.datasource.DataSourceUtils.doGetConnection(DataSourceUtils.java:115)
	at org.springframework.jdbc.datasource.DataSourceUtils.getConnection(DataSourceUtils.java:78)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:371)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at varausjarjestelma.VarausDao.list(VarausDao.java:66)
	at varausjarjestelma.HuoneDao.haeHuoneita(HuoneDao.java:69)
	at varausjarjestelma.Tekstikayttoliittyma.haeHuoneita(Tekstikayttoliittyma.java:103)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:51)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
Caused by: org.h2.jdbc.JdbcSQLException: General error: "java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]" [50000-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	... 46 more
Caused by: java.lang.IllegalStateException: The file is locked: nio:C:/Users/OMISTAJA/Documents/NetBeansProjects/mooc-tikape-kesa-19-projekti/tikape_Varausjarjestelma/hotelliketju.mv.db [1.4.197/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:870)
	at org.h2.mvstore.FileStore.open(FileStore.java:173)
	at org.h2.mvstore.MVStore.<init>(MVStore.java:350)
	... 40 more
2019-07-16 10:47:52 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'' (LISAVARUSTE: VARAUS_ID INTEGER)"; SQL statement:
INSERT INTO "PUBLIC"."LISAVARUSTE"("ID","VARAUS_ID","NIMI","PAIVAHINTA")VALUES(?,?,?,?) -- (?1, ?2, ?3, ?4) [22018-197]
2019-07-16 10:47:52 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Data conversion error converting "'' (LISAVARUSTE: VARAUS_ID INTEGER)"; SQL statement:
INSERT INTO "PUBLIC"."LISAVARUSTE"("ID","VARAUS_ID","NIMI","PAIVAHINTA")VALUES(?,?,?,?) -- (?1, ?2, ?3, ?4) [22018-197]
2019-07-18 22:36:23 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "HUONENRO" not found; SQL statement:
insert into huonevaraus(varausid, huonenro) values(25, 301) [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.table.Table.getColumn(Table.java:682)
	at org.h2.command.Parser.parseColumn(Parser.java:936)
	at org.h2.command.Parser.parseColumnList(Parser.java:920)
	at org.h2.command.Parser.parseInsertGivenTable(Parser.java:1258)
	at org.h2.command.Parser.parseInsert(Parser.java:1222)
	at org.h2.command.Parser.parsePrepared(Parser.java:431)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-07-18 22:41:02 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO HUONEVARAUS(VARAUSID, HUONENUMERO) VALUES(1, 302)
INSERT[*] INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKASID) VALUES(2019-10-20, 2019-10-30, 25) "; SQL statement:
insert into huonevaraus(varausid, huonenumero) values(1, 302)
insert into varaus(alkupvm, loppupvm, asiakasid) values(2019-10-20, 2019-10-30, 25) [42000-197]
2019-07-18 22:41:50 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Cannot parse "DATE" constant "1989"; SQL statement:
insert into varaus(alkupvm, loppupvm, asiakasid) values(2019-10-20, 2019-10-30, 25) -- (1989, 1979, 25) [22007-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:168)
	at org.h2.value.ValueDate.parse(ValueDate.java:74)
	at org.h2.value.Value.convertTo(Value.java:1069)
	at org.h2.table.Column.convert(Column.java:177)
	at org.h2.command.dml.Insert.insertRows(Insert.java:166)
	at org.h2.command.dml.Insert.update(Insert.java:134)
	at org.h2.command.CommandContainer.update(CommandContainer.java:102)
	at org.h2.command.Command.executeUpdate(Command.java:261)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:233)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: 1989
	at org.h2.util.DateTimeUtils.parseDateValue(DateTimeUtils.java:345)
	at org.h2.value.ValueDate.parse(ValueDate.java:72)
	... 15 more
2019-07-18 22:43:48 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO HUONEVARAUS(VARAUSID, HUONENUMERO) VALUES(1, 301)
INSERT[*] INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKASID) VALUES(2019-11-20, 2019-11-30, 25) "; SQL statement:
insert into huonevaraus(varausid, huonenumero) values(1, 301)
insert into varaus(alkupvm, loppupvm, asiakasid) values(2019-11-20, 2019-11-30, 25) [42000-197]
2019-07-18 22:44:02 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO HUONEVARAUS(VARAUSID, HUONENUMERO) VALUES(1, 301)
INSERT[*] INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKASID) VALUES('2019-11-20', '2019-11-30', 25) "; SQL statement:
insert into huonevaraus(varausid, huonenumero) values(1, 301)
insert into varaus(alkupvm, loppupvm, asiakasid) values('2019-11-20', '2019-11-30', 25) [42000-197]
2019-07-18 22:44:33 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO HUONEVARAUS(VARAUSID, HUONENUMERO) VALUES(1, 301)
INSERT[*] INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKASID) VALUES('2019-11-20', '2019-11-30', 2) "; SQL statement:
insert into huonevaraus(varausid, huonenumero) values(1, 301)
insert into varaus(alkupvm, loppupvm, asiakasid) values('2019-11-20', '2019-11-30', 2) [42000-197]
2019-07-18 22:44:51 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "INSERT INTO HUONEVARAUS(VARAUSID, HUONENUMERO) VALUES(1, 301)
INSERT[*] INTO VARAUS(ALKUPVM, LOPPUPVM, ASIAKASID) VALUES('2019-12-20', '2019-12-30', 3) "; SQL statement:
insert into huonevaraus(varausid, huonenumero) values(1, 301)
insert into varaus(alkupvm, loppupvm, asiakasid) values('2019-12-20', '2019-12-30', 3) [42000-197]
2019-07-18 23:26:03 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column count does not match; SQL statement:
INSERT INTO Varaus(alkupvm, loppupvm, asiakasid) VALUES(?,?,?,?) [21002-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.message.DbException.get(DbException.java:144)
	at org.h2.command.dml.Insert.prepare(Insert.java:306)
	at org.h2.command.Parser.prepareCommand(Parser.java:283)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:1166)
	at com.zaxxer.hikari.pool.ProxyConnection.prepareStatement(ProxyConnection.java:318)
	at com.zaxxer.hikari.pool.HikariProxyConnection.prepareStatement(HikariProxyConnection.java)
	at varausjarjestelma.VarausDao.lambda$createAndReturnKey$1(VarausDao.java:76)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:615)
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:893)
	at varausjarjestelma.VarausDao.createAndReturnKey(VarausDao.java:75)
	at varausjarjestelma.Tekstikayttoliittyma.lisaaVaraus(Tekstikayttoliittyma.java:160)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:54)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
2019-07-19 10:10:42 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Ambiguous column name "ID"; SQL statement:
SELECT alkupvm, loppupvm FROM Varaus
JOIN Asiakas ON id = asiakasid
WHERE id = 1 [90059-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.mapColumn(ExpressionColumn.java:127)
	at org.h2.expression.ExpressionColumn.mapColumns(ExpressionColumn.java:97)
	at org.h2.expression.Comparison.mapColumns(Comparison.java:510)
	at org.h2.table.TableFilter$MapColumnsVisitor.accept(TableFilter.java:1228)
	at org.h2.table.TableFilter.visit(TableFilter.java:1182)
	at org.h2.table.TableFilter.addJoin(TableFilter.java:669)
	at org.h2.command.Parser.addJoin(Parser.java:1857)
	at org.h2.command.Parser.readJoin(Parser.java:1793)
	at org.h2.command.Parser.parseJoinTableFilter(Parser.java:2183)
	at org.h2.command.Parser.parseSelectSimpleFromPart(Parser.java:2139)
	at org.h2.command.Parser.parseSelectSimple(Parser.java:2287)
	at org.h2.command.Parser.parseSelectSub(Parser.java:2133)
	at org.h2.command.Parser.parseSelectUnion(Parser.java:1946)
	at org.h2.command.Parser.parseSelect(Parser.java:1919)
	at org.h2.command.Parser.parsePrepared(Parser.java:463)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-07-19 10:10:57 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT ALKUPVM, LOPPUPVM FROM VARAUS
JOIN ASIAKAS ONASIAKAS.[*]ID = VARAUS.ASIAKASID
WHERE ID = 1 "; SQL statement:
SELECT alkupvm, loppupvm FROM Varaus
JOIN Asiakas ONAsiakas.id = Varaus.asiakasid
WHERE id = 1 [42000-197]
2019-07-19 10:11:05 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Ambiguous column name "ID"; SQL statement:
SELECT alkupvm, loppupvm FROM Varaus
JOIN Asiakas ON Asiakas.id = Varaus.asiakasid
WHERE id = 1 [90059-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.expression.ExpressionColumn.mapColumn(ExpressionColumn.java:127)
	at org.h2.expression.ExpressionColumn.mapColumns(ExpressionColumn.java:97)
	at org.h2.expression.Comparison.mapColumns(Comparison.java:510)
	at org.h2.expression.ConditionAndOr.mapColumns(ConditionAndOr.java:260)
	at org.h2.command.dml.Select.mapColumns(Select.java:1240)
	at org.h2.command.dml.Select.init(Select.java:826)
	at org.h2.command.Parser.parseSelect(Parser.java:1925)
	at org.h2.command.Parser.parsePrepared(Parser.java:463)
	at org.h2.command.Parser.parse(Parser.java:335)
	at org.h2.command.Parser.parse(Parser.java:307)
	at org.h2.command.Parser.prepareCommand(Parser.java:278)
	at org.h2.engine.Session.prepareLocal(Session.java:611)
	at org.h2.engine.Session.prepareCommand(Session.java:549)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:1077)
	at org.h2.server.web.WebApp.getResult(WebApp.java:1386)
	at org.h2.server.web.WebApp.query(WebApp.java:1060)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1022)
	at org.h2.server.web.WebApp$1.next(WebApp.java:1009)
	at org.h2.server.web.WebThread.process(WebThread.java:165)
	at org.h2.server.web.WebThread.run(WebThread.java:90)
	at java.lang.Thread.run(Unknown Source)
2019-07-19 23:38:53 jdbc[3]: exception
org.h2.jdbc.JdbcSQLException: Column "huoneenTyyppi" not found [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:357)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.jdbc.JdbcResultSet.getColumnIndex(JdbcResultSet.java:3148)
	at org.h2.jdbc.JdbcResultSet.get(JdbcResultSet.java:3247)
	at org.h2.jdbc.JdbcResultSet.getString(JdbcResultSet.java:310)
	at com.zaxxer.hikari.pool.HikariProxyResultSet.getString(HikariProxyResultSet.java)
	at varausjarjestelma.Varaustiedot.<init>(Varaustiedot.java:32)
	at varausjarjestelma.VarausDao.lambda$listaaVaraustiedot$2(VarausDao.java:100)
	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:94)
	at org.springframework.jdbc.core.RowMapperResultSetExtractor.extractData(RowMapperResultSetExtractor.java:61)
	at org.springframework.jdbc.core.JdbcTemplate$1QueryStatementCallback.doInStatement(JdbcTemplate.java:440)
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:376)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:452)
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:462)
	at org.springframework.jdbc.core.JdbcTemplate.queryForObject(JdbcTemplate.java:473)
	at varausjarjestelma.VarausDao.listaaVaraustiedot(VarausDao.java:94)
	at varausjarjestelma.Tekstikayttoliittyma.listaaVaraukset(Tekstikayttoliittyma.java:174)
	at varausjarjestelma.Tekstikayttoliittyma.kaynnista(Tekstikayttoliittyma.java:56)
	at varausjarjestelma.VarausjarjestelmaSovellus.run(VarausjarjestelmaSovellus.java:28)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:813)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:797)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:324)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1260)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1248)
	at varausjarjestelma.VarausjarjestelmaSovellus.main(VarausjarjestelmaSovellus.java:18)
